File: solutions/javascript/rna-transcription/1/rna-transcription.js
export const toRna = (dna) => {
  return [...dna].reduce(
    (rna, nucleotide) => rna + dnaToRnaDict[nucleotide],
    ""
  );
};

const dnaToRnaDict = {
  G: "C",
  C: "G",
  T: "A",
  A: "U",
};

File: solutions/r/rna-transcription/1/rna-transcription.R
to_rna <- function(dna) {
  nucleotides <- dna  |>
    strsplit("")  |> 
    unlist()  
  
  ifelse(test = all(nucleotides %in% names(dna_rna)),
         yes = paste(dna_rna[nucleotides], collapse=""),
         no = stop("Invalid nucleotide"))
}

dna_rna <- c("G" = "C", "C" = "G", "T" = "A", "A" = "U")
